crates/phala-types/src/contract.rs

```diff
@@ -60,13 +60,20 @@ pub mod messaging {
 #[derive(Encode, Decode, Clone, PartialEq, Eq, Debug, TypeInfo)]
 pub struct ContractInfo<CodeHash, AccountId> {
     pub deployer: AccountId,
+    pub code_index: CodeIndex<CodeHash>,
+    pub salt: Vec<u8>,
     /// Contract cluster counter of the contract
     pub cluster_id: u64,
-    pub salt: Vec<u8>,
-    pub code_index: CodeIndex<CodeHash>,
     pub instantiate_data: Vec<u8>,
 }
 
+impl<CodeHash, AccountId> ContractInfo<CodeHash, AccountId> {
+    pub fn contract_id(&self) -> ContractId {
+        // TODO.shelven: calculate the real contract id
+        id256(0)
```

Comment:
This depends on the #596 to provide the ContractInfo -> ContractId calculation.

Reply:
Time to complete it?

Reply:
Done